src/ray/gcs/gcs_server/gcs_task_manager.cc

```diff
@@ -28,6 +28,56 @@ void GcsTaskManager::Stop() {
   }
 }
 
+bool GcsTaskManager::GcsTaskManagerStorage::ShouldMarkChildrenFailed(
+    const TaskID &child_task, const TaskID &parent_task, int64_t *failed_ts) const {
+  const auto &child_task_event = GetLatestTaskEvent(child_task);
+  if (!child_task_event.has_value()) {
+    return false;
+  }
+  // TODO(rickyx): Fix detached actor
+  // // A detached actor task should not be affected by its parent's tasks.
+  // if (child_task_event->has_task_info() &&
+  //     child_task_event->task_info().is_detached_actor()) {
+  //   return false;
+  // }
+
+  // If the child task has already finished/failed, we don't need to mark it
+  if (child_task_event->has_state_updates() &&
+      (child_task_event->state_updates().failed_ts() != 0 ||
+       child_task_event->state_updates().finished_ts() != 0)) {
+    return false;
+  }
+
+  if (parent_task.IsNil()) {
+    return false;
+  }
+
+  const auto &parent_task_event = GetLatestTaskEvent(parent_task);
+
+  if (!parent_task_event.has_value()) {
+    return false;
+  }
+
+  // Check if parent has failed.
+  if (!parent_task_event->has_state_updates()) {
```

Comment:
when is this empty? isn't this always supposed to be there?

Reply:
this could be empty when the profling events are reported first. will add a comment.